src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,24 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            return list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
```

Comment:
To ensure thread safety and consistent reads, consider acquiring a read lock here. You can do this by wrapping the return statement with `lock.readLock().lock()` and `lock.readLock().unlock()` in a try-finally block, similar to the `get` method.